### PR TITLE
Implemented Issue #258

### PR DESCRIPTION
**Pull request by aa15032261**

### CODE REVIEW

**Comment by maybegreat48** [comment]
Am a bit curious on how the huge amount of TSE hashes will be updated. I generally have a comment with some code/strings that will not change for non-public events. Also this will protect against the broadcast array size overwrite exploit someone uploaded to github a couple of months ago

**Comment by DaniTRDev** [comment]
> Am a bit curious on how the huge amount of TSE hashes will be updated. I generally have a comment with some code/strings that will not change for non-public events. Also this will protect against the broadcast array size overwrite exploit someone uploaded to github a couple of months ago

Through pattern scanning inside scripts, I've made a script disassembler & assembler that can generate good signatures(it needs to be updated since I didn't add the new opcode bittest).
If anyone wants it I can make the repo public.

**Comment by maybegreat48** [comment]
I made something similar [here](https://github.com/maybegreat48/ScrUpdate) but script events are not supported for now. Perhaps some kind of xref and switch case finder might allow that to be implemented

**Comment by DaniTRDev** [comment]
Mine can be more or less adapted, I'll post the link here in a few minutes.

> I made something similar here <https://github.com/maybegreat48/ScrUpdate>
> but script events are not supported for now. Perhaps some kind of xref and
> switch case finder might allow that to be implemented
>

**Comment by DaniTRDev** [comment]
I've been busy until now. Here's the [link](https://github.com/DaniTRDev/EzScriptCore) to the public repo.

File: BigBaseV2/src/hooks/script_event_handler.cpp [hunk 1]
```diff
@@ -6,7 +6,7 @@ namespace big
 	void format_string(std::string_view player_name, std::string_view protection_type, bool should_log, bool should_notify)
 	{
 		if (should_log)
-			LOG(WARNING) << "BLOCKED_SCRIPT_EVENT";
+			LOG(WARNING) << fmt::format("BLOCKED_SCRIPT_EVENT From: {} Event Type: {}", player_name.data(), protection_type.data());
```
Comment (Yimura) [inline BigBaseV2/src/hooks/script_event_handler.cpp]: Couldn't this be changed to
```cpp
LOG(WARNING) << "BLOCKED_SCRIPT_EVENT From: " << player_name << " Event Type: " << protection_type;
```

**Review by Yimura — CHANGES_REQUESTED** [review]
Overall looks good, it appears that the previous TSE crashes were incorrectly implemented as well (on their notification part), seems I missed that and thanks for fixing that in this PR.

File: BigBaseV2/src/hooks/script_event_handler.cpp (hunk 1 above)
Comment (aa15032261) [reply BigBaseV2/src/hooks/script_event_handler.cpp]: Should've done this, don't know what I was thinking

**Review by Yimura — APPROVED** [review]

**Pull request merged**